core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java

```diff
@@ -15,67 +15,32 @@
  */
 package com.datastax.oss.driver.internal.core.os;
 
-import com.datastax.oss.driver.internal.core.util.Reflection;
-import java.lang.reflect.Method;
-import jnr.ffi.LibraryLoader;
 import jnr.ffi.Platform;
```

Comment:
this class cannot have any dependency on `jnr.ffi`, otherwise quarkus native image build will fail. We need to remove this dependency.

Reply:
Platform is a different case from the other jnr-ffi code.  It doesn't actually execute any native calls... it just builds up some state based on sysprops and exposes that.  I think there's a comment or two to this effect at https://github.com/absurdfarce/graal-cassandra-driver.

Regardless, we probably can expose this information in some other way.  I'll dig into the Platform code a bit to see what it's looking at to populate this data.  If we can pull the info from the same place and just include a default impl in the NativeImpl interface that would be awesome. :)

Reply:
I wound up folding the jnr.ffi.Platform code into Native and doing away with our dependence on Platform all together.  As mentioned above our usage of Platform is really just an eval of a sysprop, so now we just do all of that inline.

Reply:
> Platform is a different case from the other jnr-ffi code. It doesn't actually execute any native calls...

Wow I never realized that. In that case, 👍 to remove it, it's not worth the extra dependency. We'll also need to update the [integration page](https://docs.datastax.com/en/developer/java-driver/4.5/manual/core/integration/#native-libraries) in the manual.

jnr-ffi is ASL so copying the code is fine. However I think it would be a good idea to reproduce their header in our code, along the lines of what I did in [CountingIterator](https://github.com/datastax/java-driver/blob/4.x/core/src/main/java/com/datastax/oss/driver/internal/core/util/CountingIterator.java#L40-L57).

Reply:
:100: agree with the idea of maximizing attribution so I'll add the jnr-ffi license as requested.  I'm also updating our docs to clearly indicate that jnr-posix rather than jnr-ffi is the guiding force behind the underlying native calls.

Note, however, that we won't wind up removing jnr-ffi in it's entirety as it's required by jnr-posix.  But we can remove our declaration of it and just let jnr-posix pull in whatever it needs.  Apparently this has come up before (JAVA-1797) with Andy landing on a similar idea.

My only concern with that approach is making sure we're okay on the shaded JAR; I'll check that before I put all this in.